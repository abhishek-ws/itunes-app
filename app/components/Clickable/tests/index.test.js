/**
 *
 * Tests for Clickable
 *
 */

import React from 'react';
import { fireEvent } from '@testing-library/dom';
import { renderWithIntl } from '@utils/testUtils';
import Clickable from '../index';

describe('<Clickable /> component tests', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<Clickable />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 Clickable component', () => {
    const { getAllByTestId } = renderWithIntl(<Clickable />);
    expect(getAllByTestId('clickable').length).toBe(1);
  });

  it('should contain render the text according to the textId', () => {
    const { getAllByText } = renderWithIntl(<Clickable textId="list_songs" />);
    expect(getAllByText(/List of songs will appear below/).length).toBe(1);
  });

  it('should call the prop onClick when the clickable component is clicked', () => {
    const clickSpy = jest.fn();
    const { getAllByText, queryByText } = renderWithIntl(<Clickable onClick={clickSpy} textId="list_songs" />);
    expect(getAllByText(/List of songs will appear below/).length).toBe(1);
    fireEvent.click(queryByText(/List of songs will appear below/));
    expect(clickSpy).toBeCalled();
  });
});
