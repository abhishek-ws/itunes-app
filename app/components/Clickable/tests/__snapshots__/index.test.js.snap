// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clickable /> component tests should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="Clickable__StyledClickable-sc-19lp5om-0 grlVIv"
      data-testid="clickable"
    >
      <p
        class="T__StyledText-gjlic1-0 kYcyRq"
        data-testid="t"
      >
        List of songs will appear below
      </p>
    </div>
  </div>
</body>
`;
